lib/go/thrift/simple_server_test.go

```diff
@@ -154,3 +158,112 @@ func TestNoHangDuringStopFromDanglingLockAcquireDuringAcceptLoop(t *testing.T) {
 	runtime.Gosched()
 	serv.Stop()
 }
+
+func TestNoHangDuringStopFromClientNoDataSendDuringAcceptLoop(t *testing.T) {
+	ln, _ := net.Listen("tcp", "localhost:0")
+
+	proc := &mockProcessor{
+		ProcessFunc: func(in, out TProtocol) (bool, TException) {
+			in.ReadMessageBegin(context.Background())
+			return false, nil
+		},
+	}
+
+	trans := &mockServerTransport{
+		ListenFunc: func() error {
+			return nil
+		},
+		AcceptFunc: func() (TTransport, error) {
+			conn, err := ln.Accept()
+			if err != nil {
+				return nil, err
+			}
+
+			return NewTSocketFromConnConf(conn, nil), nil
+		},
+		CloseFunc: func() error {
+			return nil
+		},
+		InterruptFunc: func() error {
+			return ln.Close()
+		},
+	}
+
+	serv := NewTSimpleServer2(proc, trans)
+	go serv.Serve()
+	time.Sleep(10 * time.Millisecond)
+
+	netConn, err := net.Dial("tcp", ln.Addr().String())
+	if err != nil || netConn == nil {
+		t.Fatal("error when dial server")
+	}
+	time.Sleep(10 * time.Millisecond)
+
+	st := time.Now()
+	ServerCloseTimeout = 50 * time.Millisecond
+	err = serv.Stop()
+	if err != nil {
+		t.Logf("error when stop server:%v", err)
+		t.FailNow()
+	}
+	if time.Since(st) < 50*time.Millisecond {
+		t.Logf("stop cost less time than socket timeout, server close timeout:%v,cost time:%v", ServerCloseTimeout, time.Since(st))
```

Comment:
>less time than socket timeout

this is actually "stop/close timeout", not "socket timeout"

Reply:
also calling `time.Since` again will give you a different value and it might be more than 50ms now and that will make the error message confusing. so do something like this instead:
```
if elapsed := time.Since(st); elapsed < 50*time.Millisecond {
  t.Errorf("...", ServerCloseTimeout, elapsed)
}
```

(also there's no reason to use `t.Logf` then `t.FailNow()`, just use `t.Errorf`)

Reply:
also we should define `50*time.Millisecond` as a function scoped constant as we are using it several times inside this function.